symbology/symbol-dictionary-sample/src/main/java/com/esri/samples/symbology/SymbolDictionaryController.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright 2015 Esri.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ * 
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.esri.samples.symbology;
+
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+
+import javafx.application.Platform;
+import javafx.event.ActionEvent;
+import javafx.fxml.FXML;
+import javafx.scene.control.ProgressIndicator;
+import javafx.scene.control.ScrollPane;
+import javafx.scene.control.TextField;
+import javafx.scene.image.Image;
+import javafx.scene.image.ImageView;
+import javafx.scene.layout.HBox;
+import javafx.scene.layout.VBox;
+import javafx.scene.text.Text;
+
+import com.esri.arcgisruntime.concurrent.ListenableFuture;
+import com.esri.arcgisruntime.geometry.Point;
+import com.esri.arcgisruntime.symbology.CimSymbol;
+import com.esri.arcgisruntime.symbology.StyleSymbolSearchParameters;
+import com.esri.arcgisruntime.symbology.StyleSymbolSearchResult;
+import com.esri.arcgisruntime.symbology.SymbolDictionary;
+
+public class SymbolDictionaryController {
+
+  // injected elements from fxml
+  @FXML private ScrollPane scrollPane;
+  @FXML private TextField nameField;
+  @FXML private TextField tagField;
+  @FXML private TextField symbolClassField;
+  @FXML private TextField categoryField;
+  @FXML private TextField keyField;
+  @FXML private Text searchResultsFound;
+  @FXML private ProgressIndicator progress;
+
+  private VBox searchResultsBox;
+
+  private SymbolDictionary dictionarySymbol;
+  private StyleSymbolSearchParameters searchParameters;
+
+  /**
+   * 
+   */
+  public void initialize() {
+    // loads a specification for the symbol dictionary
+    dictionarySymbol = new SymbolDictionary("mil2525d");
+    dictionarySymbol.loadAsync();
+
+    // parameters used to search for symbol dictionary
+    searchParameters = new StyleSymbolSearchParameters();
+  }
+
+  /**
+   * Searches through the symbol dictionary using the text from the search fields.
+   * 
+   * @param event action from search button
+   */
+  @FXML
+  private void handleSearchAction(ActionEvent event) {
+    // accessing text from all search fields 
+    searchParameters.getNames().add(nameField.getText());
+    searchParameters.getTags().add(tagField.getText());
+    searchParameters.getSymbolClasses().add(symbolClassField.getText());
+    searchParameters.getCategories().add(categoryField.getText());
+    searchParameters.getKeys().add(keyField.getText());
+
+    new Thread(() -> {
+      try {
+        progress.setVisible(true);

```

Comment:
@tyle8552 Here is where I added the progress indicator, I created thread to make the sample run faster. Let me know if I am over doing it.